src/aabb.rs

```diff
@@ -141,6 +139,24 @@ pub trait Aabb<
         let max = P::from_vec(self.max().to_vec().mul_element_wise(v));
         Aabb::new(min, max)
     }
+
+    /// Give a list of the corner vertices of this bounding box.
+    fn corners(&self) -> Vec<P>;
+
+    /// Apply an arbitrary transform to the corners of this bounding box,
+    /// return a new conservative bound.
+    fn transform<T>(&self, transform: &T) -> Self
+    where
+        T: Transform<P>,
+    {
+        let base = Self::new(
+            transform.transform_point(self.min()),
+            transform.transform_point(self.max()),
+        );
+        self.corners().iter().fold(base, |u, &corner| {
```

Comment:
another problem with having `transform` and `corners` in the trait is the dynamic allocation of the corners `Vec`, which is undesirable for geometry computations. One way to address this would be to replace by a `SmallVec` with a reasonable limit, although I'd like to understand first if having the function implementation in the trait is needed in the first place. Do you have a use case?